src/libpmemobj/heap.c

```diff
@@ -103,28 +103,35 @@ heap_alloc_classes(struct palloc_heap *heap)
 	return heap->rt->alloc_classes;
 }
 
-
 /*
- * heap_arena_init -- (internal) initializes arena instance
+ * heap_arena_new -- (internal) initializes arena instance
  */
-static void
-heap_arena_init(struct arena *arena)
+static struct arena *
+heap_arena_new()
 {
+	struct arena *arena = Malloc(sizeof(struct arena));
+	if (arena == NULL) {
+		ERR("heap: arena malloc error");
+		return NULL;
+	}
 	arena->nthreads = 0;
 
 	for (int i = 0; i < MAX_ALLOCATION_CLASSES; ++i)
 		arena->buckets[i] = NULL;
+
```

Comment:
you will need to actually allocate those buckets, and the appropriate recycler. See `heap_create_alloc_class_buckets`.